packages/core/src/renderer/components/kube-object-details/kube-object-detail-items/metrics/getMetricsKubeObjectDetailItem.injectable.tsx

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import type { KubeObjectDetailMetrics } from "@k8slens/metrics";
+import type { InjectionToken } from "@ogre-tools/injectable";
+import { getInjectable } from "@ogre-tools/injectable";
+import type { ClusterMetricsResourceType } from "../../../../../common/cluster-types";
+import metricsDetailsComponentEnabledInjectable from "../../../../api/catalog/entity/metrics-details-component-enabled.injectable";
+import type { KubeObjectDetailItem } from "../kube-object-detail-item-injection-token";
+import { EmptyMetrics } from "./empty-metrics";
+
+type GetMetricsKubeObjectDetailItem = (token: InjectionToken<KubeObjectDetailMetrics, void>, metricResourceType: ClusterMetricsResourceType) => KubeObjectDetailItem;
+
+export const getMetricsKubeObjectDetailItemInjectable = getInjectable({
```

Comment:
Does this have to be a factory function? Eg. instead of `getMetricsKubeObjectDetailItemInjectable`, could it be `metricsKubeObjectDetailItemInjectable`, and receive `metricResourceType` as its key as `keyedSingleton`?